plugins/get_url/komand_get_url/actions/get_file/action.py

```diff
@@ -15,69 +13,45 @@ def __init__(self):
             input=GetFileInput(),
             output=GetFileOutput(),
         )
+        self.utils = Utils(action=self)
 
     def run(self, params={}):
-        utils = Utils(action=self)
         url = params.get(Input.URL)
         checksum = params.get(Input.CHECKSUM)
-        tout = params.get(Input.TIMEOUT, 60)
+        timeout = params.get(Input.TIMEOUT, 60)
         is_verify = params.get(Input.IS_VERIFY, True)
-
-        # Check for supported url prefix
-        utils.validate_url(url)
-
-        meta = utils.hash_url(url)
-        cache_file = "/var/cache/" + meta["file"]
-
-        # Attempt to retrieve headers from past request
-        headers = {}
-        if komand.helper.check_cachefile(meta["metafile"]):
-            headers = utils.check_url_meta_file(meta)
-
-        # Download file
-        url_object = komand.helper.open_url(
-            url,
-            timeout=tout,
-            verify=is_verify,
-            If_None_Match=headers.get("etag", ""),
-            If_Modified_Since=headers.get("last-modified", ""),
-            User_Agent=params.get(Input.USER_AGENT, "Mozilla/5.0"),
-        )
-
+        user_agent = params.get(Input.USER_AGENT, "Mozilla/5.0")
+        url_object, meta = self.utils.check_prefix_and_download(url, is_verify, user_agent, timeout)
+        cache_file = "/var/cache/" + meta.get("file")
         if url_object:
-            contents = url_object.read(8388608).decode("utf-8")
-
+            contents = url_object.read().decode("utf-8", "replace")
             # Optional integrity check of file
-            if checksum:
-                if not komand.helper.check_hashes(contents, checksum):
-                    self.logger.error("GetFile: File Checksum Failed")
-                    raise PluginException(cause="GetURL Failed", assistance="File Checksum Failed")
+            if checksum and not komand.helper.check_hashes(contents, checksum):
+                self.logger.error("GetFile: File Checksum Failed")
+                raise PluginException(cause="GetURL Failed", assistance="File Checksum Failed")
```

Comment:
Could we include the mismatching expected/result checksums in the assistance error message? Something like `assistance=f"File checksum {result} does not match {expected}."`